build_tools/cmake/iree_llvm.cmake

```diff
@@ -77,7 +77,34 @@ macro(iree_llvm_configure_bundled)
   set(IREE_LLVM_LINK_BINARY "$<TARGET_FILE:${IREE_LLVM_LINK_TARGET}>")
   set(IREE_LLD_BINARY "$<TARGET_FILE:${IREE_LLD_TARGET}>")
   set(IREE_CLANG_BINARY "$<TARGET_FILE:${IREE_CLANG_TARGET}>")
-  set(IREE_CLANG_BUILTIN_HEADERS_PATH "${LLVM_BINARY_DIR}/lib/clang/${CLANG_EXECUTABLE_VERSION}/include/")
+
+  # Find the clang builtin headers path. It is always of the form
+  #   ${LLVM_BINARY_DIR}/lib/clang/${LLVM_VERSION_MAJOR}/include/
+  # so this task is equivalent to finding the value of ${LLVM_VERSION_MAJOR}.
+  # We can't simply read ${LLVM_VERSION_MAJOR} here, because:
+  # - The LLVM CMake setup only defines it for its own use, but does not export it.
```

Comment:
Discussed offline.

* We'll see if LLVM would accept a change to export `LLVM_VERSION_MAJOR` (might just be `set(PARENT_SCOPE ...)`)
* Bazel using 'staging' instead of a specific version. Would be nice if we could do the same in CMake: https://github.com/openxla/iree/blob/32113b0f16e56f9e5909cf1c266fa300603ff1e7/build_tools/bazel/iree_bitcode_library.bzl#L70
* If all else fails, this is brittle but at least isolated to a file filled with other brittle LLVM-y things. Might be fine to let it slide

Reply:
Filed https://github.com/llvm/llvm-project/pull/83346.